### DISCUSSION TITLE
Linux 6.9.x

### DISCUSSION THREAD

**Discussion by zarMarco** [openzfs/zfs#16363] [General]
Hi all, when upgrade for linux 6.9? I see that 6.10 is out

**Comment by IvanVolosyuk** [comment]
Never? ZFS always lags behind the latest Linux release, so the safe upgrade path is to use the latest LTS kernel. Alternatively you can find yourself in situation where your kernel is no longer supported by kernel developers, but the next one is not supported by ZFS yet. Goodbye security updates.

**Reply by zarMarco** [reply]
So, you don't recommend using it?

**Reply by IvanVolosyuk** [reply]
There are multiple levels of tolerance to instability. Everyone chooses for themselves.

**Comment by robn** [comment]
2.2.5 should land in the next week or two, and unless something changes, will have support for 6.9 and experimental support for 6.10 (both have been supported on the development branch for a little while now).

**Reply by zarMarco** [reply]
Thanks so much!

**Reply by n0xena** [reply]
I noticed the 2.2.5 patchset PR failed with only minor issues
will these delay 2.2.5 and with it 6.9/6.10? or do you see the project stable enough for the current kernel?

**Reply by thulle** [reply]
FWIW I threw the current PR onto 2.2.4 2 days ago to be able to try kernel 6.10, and everything has been fine so far. I run modules + initramfs though, if you're compiling ZFS into the kernel you prob need 16390.

**Comment by n0xena** [comment]
so the 2.2.5 PR completed all tests successfull - does that mean zfs might be ready for 6.9/6.10?
if so - although better to wait for an official release - is there some way I could try this myself locally? or with cloning the repo and just merging it?

**Comment by gentoosys** [comment]
Waiting for 2.2.5 to upgrade my kernel to 6.10. I love zfs, but this kernel update delay is awful. Hope some day ZFS will be merged in kernel (there are firmware blobs, why not include CDDL code?)

**Reply by mskarbek** [reply]
It is what it is, and it will not change in the forceable future. Asking this question over and over will not make it suddenly possible. If you use and, as you said, love ZFS, then you should accept this fact. Demanding latest kernel compatibility at the kernel release day is unrealistic.

**Reply by fabrica64** [reply]
For me the problem is updating from Fedora 39 to 40.
I am on kernel 6.8 but if I use system-upgrade it will upgrade using kernel 6.9 (and 40 will soon use 6.10) disabling ZFS.
I hope being able to do the upgrade before Fedora EoL in november :-)

**Reply by n0xena** [reply]
because GPL (kernel) and CDDL (zfs) are incompatible with eachother

**Comment by tonyhutter** [comment]
zfs-2.2.5 was just released with 6.9 kernel support: https://github.com/openzfs/zfs/releases/tag/zfs-2.2.5

**Reply by zarMarco** [reply]
Now I'm waiting for compatibility for 6.10 ihihih
